lib/ansible/modules/cloud/misc/proxmox_kvm.py

```diff
@@ -655,6 +742,32 @@ def get_vminfo(module, proxmox, node, vmid, **kwargs):
     results['vmid'] = int(vmid)
 
 
+def get_vm_agent_info(proxmox, node, vmid, agent_info):
+    info = {}
+    warn = []
+    proxmox_vm_agent = getattr(proxmox.nodes(node), VZ_TYPE)(vmid).agent
+    for info_key in agent_info:
+        try:
+            info[info_key] = proxmox_vm_agent.get(get_agent_info_api_endpoint(info_key))['result']
+        except Exception as e:
+            if isinstance(e.args, tuple) and len(e.args) > 0:
+                warn.append('Agent info not available for \'%s\': %s' % (info_key, get_agent_info_api_error_message(e.args)))
+            else:
+                warn.append('Not able to get agent info for \'%s\', unkown error.' % info_key)
+    return info, warn
+
+
+def get_agent_info_api_endpoint(info_key):
+    return AGENT_INFO_API_MAP[info_key] if info_key in AGENT_INFO_API_MAP else info_key
+
+
+def get_agent_info_api_error_message(error_tuple):
+    error, = error_tuple
```

Comment:
How about
```suggestion
    error = error_tuple[0]
```
That makes it look less like a typo :)